### DISCUSSION TITLE
这两天遇到的一个突然几分钟不通，然后自己恢复，又反复的现象

### DISCUSSION THREAD

**Discussion by chika0801** [XTLS/Xray-core#1772] [General]
类似现象相关 Issues

https://github.com/3andne/restls/issues/8

https://github.com/XIU2/CloudflareSpeedTest/issues/217

---

vps是virmach荷兰，闲置3个多月。二手机IP前科怎样不清楚。国际回程线路。

昨天DD了新系统，使用vltss+vision+tls，端口443，域名没用全新的，一个旧域名，添加一个子域名指向VPS的IP，用cerbot申请证书，CA是le的。

客户端节点信息，地址填域名和地址填IP，SNI填域名都有测试。

现在昨天和今天都观察到了。本机在v2rayN选这个节点，网页第1次打开了，过几秒网页打不开。

这时本机icmp ping VPS和用tcp ping 443 VPS ， 观察到icmp ping一直通。tcp ping超时，此时同时去ping.pe检查443，显示CN和CN外是绿色。

tcp ping不通约3分钟，自己恢复。此时我不用这节点，本机一直tcp ping VPS，等了1-2分钟 ，tcp ping又中断，中断再等几分钟又恢复。我又选到这节点，继续看网页，又没复现。

做一个记录。这现象，在另1个国际线路回程的VPS（法兰克福）上也遇到过1回，这几天的时间段发现的。没有留意观察过更早时间点（这些VPS平时没用）。因为平时也不用这些机，后续可能没更新信息。

主力用的VPS也是vltss+vision+tls 443 回程优化线路，没有观察到类似现象。

**Comment by chika0801** [comment]
我等了5-10分钟后，又切换回这节点，到现在持续使用了15分钟，此现象没复现。

2台VPS，VPS上用wgcf配置，wireguard全局进warp。发生前，这2个VPS本机ping一切正常。切换到VPS法兰克福，访问 https://ip.skk.moe/ 本来想查看下IP。发现网页打不开，此时查看本机的tcp ping VPS 中断。马上切到荷兰，还是访问  ip.skk.moe 现象复现。等了2-3分钟，现在2台机，从本机tcp ping看恢复了，切换到这2个节点 访问网页又正常。VPS法兰克福是刚重新DD系统，使用的是Vision+Reality。荷兰是Vision+tls。

再过了约10分钟，找了台vultr的法兰克福用Vision+Reality，也安好warp。本地选到这节点，访问网页，没有被中断。切换到其它2个VPS节点，也正常。（记录下）

vultr的法兰克福再用了10来分钟，突然打不开网页，查看本机tcp ping超时，切到其它节点使用。等了约2分钟tcp ping看到恢复，代理正常了。切到vir荷兰的节点，同样用10来分钟，复现不通了(2分钟后恢复)。恢复后几个切点分别用了5-10分钟没复现，就没试了。不好说原因，线路质量我认为可能性大点。
----------------------以上是第1天深夜3-5点----------------------

第1天下午2点，这3个节点轮流使用，网页1个youtube放视频，再浏览其它网页，每个节点使用了3-5分钟，3个节点全和上面一样的现象出现。然后更换到回程优化的VPS，用的vision+tls，网页放视频，没复现。等了10来分钟，切回这3个节点轮流用，都又没有复现。

v2rayN的分流规则是

```
        "domainStrategy": "AsIs",
        "rules": [
            {
                "type": "field",
                "domain": [
                    "geosite:category-ads-all"
                ],
                "outboundTag": "block"
            },
            {
                "type": "field",
                "domain": [
                    "geosite:category-games@cn"
                ],
                "outboundTag": "direct"
            },
            {
                "type": "field",
                "domain": [
                    "geosite:geolocation-!cn"
                ],
                "outboundTag": "proxy"
            },
            {
                "type": "field",
                "domain": [
                    "geosite:cn",
                    "geosite:private"
                ],
                "outboundTag": "direct"
            },
            {
                "type": "field",
                "ip": [
                    "geoip:cn",
                    "geoip:private"
                ],
                "outboundTag": "direct"
            }
        ]
```

服务端规则都有

```
    "routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "ip": [
                    "geoip:cn"
                ],
                "outboundTag": "block"
            }
        ]
    },

```

**Comment by chika0801** [comment]
第2天下午3点，vir法兰克福使用Vision+reality 443复现，端口5000复现，使用trojan+tcp+tls 端口5001复现。

vultr法兰克福使用Vision+reality 端口5000复现。

应该不会再做继续更新了，这现象做个记录。

**Comment by GreatMichaelLee** [comment]
就是墙对你的IP做了SNI阻断了，我的vultr的vps一样有这个现象，IP’脏' 了，我的其他服务商的vps没任何问题。

**Reply by chika0801** [reply]
> 就是墙对你的IP做了SNI阻断了，我的vultr的vps一样有这个现象，IP’脏' 了，我的其他服务商的vps没任何问题。

刚买了一个洛杉矶cn2回程的商家的机，新系统搭好 trojan tcp tls 10001端口，youtube放网页放了15分钟，没遇到上面现象。

排除线路波动，我也更相信是IP前科“脏”

**Reply by chika0801** [reply]
我还注意到，测试是用的家宽，当发生问题后，手机用5G（同一个运营商，比如家宽电信，手机5G电信），手机是通的。手机我没花时间长时间用如10几分钟，来测会不会复现现象，没测试。

**Comment by Alnovekio** [comment]
我也出现了这样的现象，vps在新加坡，情况和你的基本一致。经过我自己的测试还有另外一些特点：
1. 这种短暂的封禁是针对端口的。我用iptable在vps上做了一个简单的端口映射，当一个端口不通了的时候换到映射的端口却能继续使用。
2. 与所用协议无关。使用Hysteria和Trojan都发生了同样的情况，尤其Hysteria还是UDP协议。
3. 与地理位置有关。当我的网络不通的时候，另一个省市的朋友却在同一时刻仍能够正常使用。

**Reply by chika0801** [reply]
感谢你分享信息

**Reply by chika0801** [reply]
3月初到3月10几日期间，遇到一些反馈断流，当时没留意，以为是用的VPS回程线路国际线路正常现象。一天看到这个 https://github.com/XTLS/Xray-core/issues/1769 我也拿一些VPS测试了下，就复现了。之前我没遇到，是因为我用的VPS都是有回程线路优化的。（并且这些VPS我到现在回复时，也没遇到过现象复现）

**Reply by Alnovekio** [reply]
我最后一次确认遇到这种现象是3月10日，在之前几乎天天都有问题。之后几天我在摆烂没怎么用网，但到了今天我再怎么使用却都复现不出来了。另外还有一点就是被封禁期间虽然换到事先映射好的端口能够正常使用，但ssh却怎么都连不上了，搞得我很混乱不知道是什么原理

**Reply by chika0801** [reply]
我找的那3台 2台vir1台vultr，然后出现现象时，ssh一直是正常的。icmp正常的。外地没朋友，当时立马换手机信号测试节点是通，家宽出现的。

**Reply by xiaosteven** [reply]
就是开会期间的sni阻断，我这段时间每天晚上基本都被阻断，上了cdn也不行，cf和cft都试过

**Reply by chika0801** [reply]
> 就是开会期间的sni阻断，我这段时间每天晚上基本都被阻断，上了cdn也不行，cf和cft都试过

今天用其中一个VPS，没遇到此现象。推测是会开完了，力度降了。

**Reply by Alnovekio** [reply]
> > 就是开会期间的sni阻断，我这段时间每天晚上基本都被阻断，上了cdn也不行，cf和cft都试过
> 
> 今天用其中一个VPS，没遇到此现象。推测是会开完了，力度降了。

我这里这几天也没问题了，应该就是上面说的那样。所以下次再发生这种情况的时候，用REALITY去偷别人会有用吗？

**Reply by chika0801** [reply]
> > > 就是开会期间的sni阻断，我这段时间每天晚上基本都被阻断，上了cdn也不行，cf和cft都试过
> > 
> > 
> > 今天用其中一个VPS，没遇到此现象。推测是会开完了，力度降了。
> 
> 我这里这几天也没问题了，应该就是上面说的那样。所以下次再发生这种情况的时候，用REALITY去偷别人会有用吗？

我测试用的3个VPS就是有用了REALITY偷别人，一样发生的。在发生期间，另外几个优化线路的VPS从未出现那现象。

**Comment by cqlp1314** [comment]
rn和甲骨文vps使用h2+reality目前扔遇到通与不通反复的现象

**Reply by chika0801** [reply]
感谢分享信息

**Reply by cqlp1314** [reply]
我最近发现不通的时候重启下客户端就立马好了，我是用grpc+reality测试的

**Reply by chika0801** [reply]
> 我最近发现不通的时候重启下客户端就立马好了，我是用grpc+reality测试的

你的这现象，是grpc/h2有名的交流时俗称的断流。特点就是遇到不通，客户端关，开，马上就恢复了。

**Reply by cqlp1314** [reply]
哦哦，有法子解决么

**Reply by chika0801** [reply]
> 哦哦，有法子解决么

没有。要不你不用GRPC/H2即可。

**Comment by Alnovekio** [comment]
3.29晚复现，情况与之前相同

**Reply by chika0801** [reply]
感谢分享信息

**Comment by sophauer** [comment]
这种现象在联通宽带高发

**Comment by sduoduo233** [comment]
我家是联通，也遇到过，也是能ping，但是tcp连不上，路由器重新拨号就能连上了。移动目前没遇到过

**Reply by JASONSE-LIN** [reply]
对 正确的做法就是重拨

**Comment by ghost** [comment]
成都移动商宽这种检测持续了几个月，开端口跳跃也没用，照样封回程，hysteria、troajn、xtls+reality， dmit、vultr、搬瓦工、olink、cloudpowerall，xxxxx 等等都试过。

**Comment by dameng9779** [comment]
5.5下午2点以后一直到现在,本地电信与联通持续出现这种现象

**Comment by chika0801** [comment]
感谢反馈信息

**Comment by wangmice** [comment]
今晚我也出现过几次，大概持续几分钟，没用reality，用的是vless，tcp-tls-vision

**Comment by gisekvl** [comment]
北京联通访问首尔甲骨文，xtls vision + reality ver.1.8.1 偷自己。443 端口 xray 一连，ICMP 整体秒断，所有其它端口都连不了。几分钟后自动解除。验证多次。只要不开xray，只访问网站，ICMP 就一直可以 ping 通。

**Reply by xcf13363175** [reply]
一样，就是ICMP是通的

**Comment by xcf13363175** [comment]
xtls vision + reality ver.1.8.1 偷自己，443 端口
相同配置电信一直稳定使用，移动会出现突然不同过两分钟恢复，ICMP通的，TCP不通。

**Comment by sophauer** [comment]
xtls vision + reality ver.1.8.1
高端口,坚持了一周,流量一大,看了几个电影,整个IP墙了...
反倒是另一台用tls+websocket没事

**Comment by qianbinbin** [comment]
我最近疑似遭遇了严重的 SNI 阻断，域名本身的网站可以正常访问，但是油管、GitHub 等网站断断续续
方案用过 VLESS+WS+TLS 和 VMess+WS+TLS，套不套 CDN 都一样

**Reply by chika0801** [reply]
> 我最近疑似遭遇了严重的 SNI 阻断，域名本身的网站可以正常访问，但是油管、GitHub 等网站断断续续 方案用过 VLESS+WS+TLS 和 VMess+WS+TLS，套不套 CDN 都一样

用 WS 组合建议看 https://github.com/chika0801/Xray-examples/blob/main/tips.md#memo-5

**Reply by qianbinbin** [reply]
非常感谢，现在有什么比较推荐的方案吗，gRPC？

**Reply by chika0801** [reply]
https://github.com/chika0801/Xray-examples

里面有4组，其实grpcH2算一种，就是3种，我用的最多的 https://github.com/chika0801/Xray-examples/tree/main/VLESS-XTLS-uTLS-REALITY/steal_yourself 这种的这形式

**Reply by qianbinbin** [reply]
多谢指点，感觉有点复杂，以后再弄吧。
神奇的是，从昨天晚些时候我就没遇到过阻断了，ssh 通过 socks 代理一直连着，之前都是几分钟断一次，烦不胜烦。。

**Comment by Smallthing** [comment]
一直想知道GRPC/H2为啥会断流，不解，如果是重启客户端马上恢复的话，为啥不能自动判断呢。。这个断流到底是什么东西导致的

**Reply by chika0801** [reply]
问GPT给的内核优化参数https://github.com/chika0801/Xray-examples/tree/main/VLESS-H2-uTLS-REALITY#%E6%B3%A8%E6%84%8F

你可以试试sing-box提供的h2mux padding 这组合

**Comment by chika0801** [comment]
一个联动，记录下
https://github.com/XTLS/Xray-core/issues/2661

**Reply by shandongtlb** [reply]
我也遇到了这个问题，dmit的机子，reality偷自己域名。
具体表现是电信5g网络客户端无法连接，ssh和ping均不通，但直接访问我的域名网站是没问题的。
同一地点的电信宽带和移动5g不会复现，目前只有电信5g会这样

**Reply by shandongtlb** [reply]
怀疑就是ip脏了，前一阵我换了一个ip，在没有安装任何服务的情况下，ping有时候不通

**Comment by cvpas** [comment]
我1.8.4也遇到了，但是回退1.8.3就好了

**Reply by chika0801** [reply]
这现象，和Xray-core版本是无关的。

**Comment by mynameiscaa** [comment]
我在江苏今天也碰到了这个情况，电信家宽之前一直都挺稳定的，用的xtls+reality，今天发现同一个地区的电信移动家宽都无法连接，但手机移动数据就可以访问，然后让外地的朋友测试了一下又都正常，不知道什么原因

**Reply by shandongtlb** [reply]
大哥你什么vps，我发现美西cn2gia线路很容易这样

**Reply by mynameiscaa** [reply]
Singapore misaka.io

**Reply by chika0801** [reply]
高端（价格）呀

**Reply by FishOrBear** [reply]
搬瓦工? cn2gia?

**Comment by Nokronstella** [comment]


**Reply by chika0801** [reply]
如果你联通随机遇到了，马上用如手机网络，如移动5g，试试是通还是不通。

你非要认为和协议组合有关，换如trojan tcp tls ，按排除法自己查了

**Reply by chika0801** [reply]
我在去年3月那几个商家机，记得我1楼写的内容当时是vultr遇到。去年后面时间用的瓦工，和一些其它优化线路商家，有1个vps只有1个人用，也有给其它人共用，没再遇到。

可能和你所在地区，运营商，vps商家各种玄学加成

**Comment by shiguangyibei** [comment]
两台机都有这个问题，感觉是联通特色，hy2断流常态不可用，reality有时会，大部分时间都是正常的
